sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -232,13 +247,35 @@ public RecordedData getRecordedData() {
         return recordedData;
     }
 
+    /**
+     * A {@link Supplier} for retrieving a variable from a test proxy recording.
+     * @return The supplier for retrieving a variable.
+     */
+    public Supplier<String> getProxyVariableSupplier() {
+        return () -> {
+            Objects.requireNonNull(this.testProxyPlaybackClient, "Playback must be started to retrieve values");
+            return proxyVariableQueue.remove();
+        };
+    }
+
+    /**
+     * Get a {@link Consumer} for adding variables used in test proxy tests.
+     * @return The consumer for adding a variable.
+     */
+    public Consumer<String> getProxyVariableConsumer() {
+        return proxyVariableQueue::add;
+    }
+
     /**
      * Gets a new HTTP pipeline policy that records network calls and its data is managed by
      * {@link InterceptorManager}.
      *
      * @return HttpPipelinePolicy to record network calls.
      */
     public HttpPipelinePolicy getRecordPolicy() {
+        if (enableTestProxy) {
+            return startProxyRecording();
```

Comment:
I'd update the Javadoc for this to mention if test proxy is enabled this won't add a recording policy

Reply:
It will, though. This is part of what makes existing tests continue to work.